lib/configuration.js

```diff
@@ -87,7 +87,7 @@ var Configuration = function(givenConfig, logger) {
    * @type {Boolean}
    */
   this._shouldReportErrorsToAPI = env.NODE_ENV === 'production';
-  if (!this._shouldReportErrorsToAPI) {
```

Comment:
In what cases would we expect `_logger` to not be defined at this point? If this is for testing only, I'd rather use a dummy logger for testing instead of adding logic to skip this warning. @cristiancavalli